src/rest_vol_attr.c

```diff
@@ -1757,40 +1764,54 @@ RV_attr_specific(void *obj, const H5VL_loc_params_t *loc_params, H5VL_attr_speci
 
                     memcpy(attr_iter_object, loc_obj, sizeof(RV_object_t));
 
+                    attr_iter_obj_typed = (RV_object_t*) attr_iter_object;
+
                     /* Since we already have the attribute's parent object, but still need an hid_t for it
                      * to pass to the user's object, we will just copy the current object, making sure to
                      * increment the ref. counts for the object's fields so that closing it at the end of
                      * this function does not close the fields themselves in the real object, such as a
                      * dataset's dataspace.
                      */
+
+                    /* Increment refs for top-level file */
+                    if (parent_obj_type == H5I_FILE    || 
+                        parent_obj_type == H5I_GROUP   || 
+                        parent_obj_type == H5I_DATASET || 
+                        parent_obj_type == H5I_DATATYPE) {
+                            
+                        loc_obj->domain->u.file.ref_count++;
+
+                        if (loc_obj->domain->u.file.fapl_id >= 0) {
+                            if ((loc_obj->domain->u.file.fapl_id != H5P_FILE_ACCESS_DEFAULT) && (H5Iinc_ref(loc_obj->domain->u.file.fapl_id) < 0))
+                                FUNC_DONE_ERROR(H5E_PLIST, H5E_CANTCLOSEOBJ, FAIL, "can't increase ref count of FAPL");
+                        } /* end if */
+
+                        if (loc_obj->domain->u.file.fcpl_id >= 0) {
+                            if ((loc_obj->domain->u.file.fcpl_id != H5P_FILE_CREATE_DEFAULT) && (H5Iinc_ref(loc_obj->domain->u.file.fcpl_id) < 0))
+                                FUNC_DONE_ERROR(H5E_PLIST, H5E_CANTCLOSEOBJ, FAIL, "can't increase ref count of FCPL");
+                        } /* end if */
```

Comment:
Is there a corresponding close call for the above? Otherwise, it looks like the memory for `loc_obj->domain->u.file` would be leaking and the IDs for `loc_obj->domain->u.file.fapl_id` and `loc_obj->domain->u.file.fcpl_id` would be leaking.

Reply:
The domain is closed inside the close call for each particular type at the end of the function. You're correct these property list IDs were leaking after the change to `RV_file_close` - fixed by removing this incrementing.